crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1381,6 +1392,133 @@ pub fn decode_calldata(data: &Bytes) -> Result<(Vec<Fulfillment>, AssessorReceip
     ))
 }
 
+#[derive(Debug, Clone)]
+/// Represents the parameters for submitting a Merkle Root.
+pub struct Root {
+    /// The address of the set verifier contract.
+    pub verifier_address: Address,
+    /// The Merkle root of the proof.
+    pub root: B256,
+    /// The seal of the proof.
+    pub seal: Bytes,
+}
+
+#[derive(Debug, Clone)]
+/// Represents the parameters for pricing a request.
+pub struct Price {
+    /// The list of requests to be priced.
+    pub requests: Vec<ProofRequest>,
+    /// The list of client signatures for the requests.
+    pub client_sigs: Vec<Bytes>,
+}
+
+#[derive(Clone)]
+#[non_exhaustive]
+/// Builder for creating a fulfillment request.
+pub struct FulfillmentBuilder<P> {
+    /// The Boundless market service instance.
+    pub boundless_market: BoundlessMarketService<P>,
```

Comment:
I don't think the `FulfillmentBuilder` (i.e. `FulfillmentTx` if you like that name) should hold an instance of the market and have a provider. Instead, we can have a function on the market that accepts a `FulfillmentBuilder` and does the work to send it.